runtime/databricks/automl_runtime/forecast/utils.py

```diff
@@ -14,14 +14,77 @@
 # limitations under the License.
 #
 import logging
-from typing import List, Optional
+from typing import Dict, List, Optional, Tuple, Union
 from databricks.automl_runtime.forecast import DATE_OFFSET_KEYWORD_MAP,\
     QUATERLY_OFFSET_ALIAS, NON_DAILY_OFFSET_ALIAS, OFFSET_ALIAS_MAP, PERIOD_ALIAS_MAP
 
 import pandas as pd
 
 _logger = logging.getLogger(__name__)
 
+def make_future_dataframe(
+        start_time: Union[pd.Timestamp, Dict[Tuple, pd.Timestamp]],
+        end_time: Union[pd.Timestamp, Dict[Tuple, pd.Timestamp]],
+        horizon: int,
+        frequency: str,
+        include_history: bool = True,
+        groups: List[Tuple] = None,
+        group_names: List[str] = None,
```

Comment:
Would it be more readable if we use a `Dict[str, Tuple]` here, which maps from group name to group (i.e. tuple)? This way we don't have to bother keeping the two lists to be same order.

Also, could we add docstr for this function? It took me a while to understand what's groups and what's group_names.